src/System.Text.Json/src/System/Text/Json/JsonReaderException.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines a custom exception object that is thrown by the <see cref="JsonUtf8Reader"/> whenever it
+    /// encounters an invalid JSON text while reading through it. This exception is also thrown
+    /// whenever you read past the defined maximum depth.
+    /// </summary>
+    public class JsonReaderException : Exception
```

Comment:
Please add an entry to BinaryFormatterTestData.cs to prove this is serialized correctly.

Reply:
Info here https://github.com/dotnet/corefx/pull/33099#discussion_r228992474
In fact it would be nice to add that as a comment to that file since it comes up often.

Reply:
The test will immediately fail because this needs `[Serializable]` .. all our exceptions should be